perceval/runtime/remote_processor.py

```diff
@@ -0,0 +1,143 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Dict, List, Callable
+
+from perceval.components.abstract_processor import AProcessor, ProcessorType
+from perceval.components import Circuit
+from perceval.utils import Parameter, BasicState, SVDistribution, generate_sync_methods
+from .remote_backend import RemoteBackend
+from .remote_job import RemoteJob
+from .rpc_handler import RPCHandler
+
+QUANDELA_CLOUD_URL = 'https://api.cloud.quandela.dev'
+
+
+def _extract_commands(specs):
+    for v in specs.values():
+        if 'available_commands' in v:
+            for c in v['available_commands']:
+                yield c
+
+
+def _get_first_spec(specs, name):
+    for v in specs.values():
+        if 'name' in v:
+            return v['name']
+    return None
+
+
+def _split_platform_and_backend_name(name: str):
+    backend_name = 'Naive'
+    platform_name = name
+    colon_pos = name.find(":")
+    if colon_pos != -1:
+        platform_name = name[:colon_pos]
+        backend_name = name[colon_pos + 1:]
+    return platform_name, backend_name
+
+
+@generate_sync_methods
+class RemoteProcessor(AProcessor):
+    def __init__(self, name: str, token: str, url: str = QUANDELA_CLOUD_URL):
+        super().__init__()
+        (self.name, self._backend_name) = _split_platform_and_backend_name(name)
+
+        self._rpc_handler = RPCHandler(self.name, url, token)
+        self._specs = {}
+        self._type = ProcessorType.SIMULATOR
+        self._circuit = None
+        self._backend: RemoteBackend = None
+        self._available_circuit_parameters = {}
+        self._input_state = None
+        self.fetch_data()
+
+    @property
+    def is_remote(self):
+        return True
+
+    def fetch_data(self):
+        platform_details = self._rpc_handler.fetch_platform_details()
+        plugins_specs = platform_details['specs']
+        self._specs.update(platform_details['specs'][next(iter(plugins_specs.keys()))])
+        if platform_details['type'] != 'simulator':
+            self._type = ProcessorType.PHYSICAL
+
+    @property
+    def specs(self):
+        return self._specs
+
+    def set_circuit(self, circuit: Circuit):
+        self._circuit = circuit
+        self.__build_backend()
+
+    def __build_backend(self):
+        if self._circuit is None:
+            raise RuntimeError("No circuit set in RemoteProcessor")
+
+        self._backend = RemoteBackend(self._rpc_handler, self._backend_name, self._circuit)
+
+    def get_rpc_handler(self):
+        return self._rpc_handler
+
+    @property
+    def type(self) -> ProcessorType:
+        return self._type
+
+    def with_input(self, input_state: BasicState) -> None:
+        self._input_state = input_state
+
+    @property
+    def available_sampling_method(self) -> str:
+        for k, v in _extract_commands(self._specs):
+            if v.startswith('sample'):
```

Comment:
I'm not sure this'll work as "probs" should be a valid answer in some cases. We should discuss this.